商品出品機能実装

5at05h1 opened the pull request:
#What
商品出品機能の実装
#Why
会員登録し、ログインしているユーザーが商品を出品できるようにする為

5at05h1 committed:
itemsテーブルの作成

5at05h1 committed:
テーブル作成

5at05h1 committed:
ルーティング設定

5at05h1 committed:
Active Storage導入

5at05h1 committed:
ActiveHash導入

5at05h1 committed:
コントローラー作成

5at05h1 committed:
ルーティング設定

5at05h1 committed:
カテゴリー選択実装１

5at05h1 committed:
カテゴリー選択実装2

kokikurahashi commented:
こちらのプルリクエストですが、実際の画面や機能ができているか確認させていただきたいので、キャプチャ動画をご提出いただけますか？
「Gyazo　GIF　使い方」とgoogleで検索し、ツールによって生成されたURLをこのPRに貼っていただければ幸いです。
お手数かけますが、よろしくお願いします。

5at05h1 commented:
https://gyazo.com/79b38f7ad51c2546cd0dee3f0dd58453

kokikurahashi commented:
こちら「商品出品」のプルリクエストでお間違い無いでしょうか？
コードを拝見しましたところnewアクションや、createアクションの記述がないためデータベースの方に商品を保存するところまでは実装できていないのではないでしょうか？
トレロにもありますように実際に商品の保存ができる(DBに保存される)ことを確認した上で単体テストを実行し、改めてコードレビューの方をお願いしてもよろしいでしょうか？

5at05h1 committed:
DB保存

5at05h1 committed:
ログインユーザー　出品ページ遷移

5at05h1 committed:
テストコード記述

chirikuzu commented on spec/models/item_spec.rb:
```diff
@@ -0,0 +1,53 @@
+        @item.price = ""
+        @item.valid?
+        expect(@item.errors.full_messages).to include("Price can't be blank")
+      end
```
テストの検証内容ですが、他にもいろんなパターンを考えてテストを行いましょう。
必要なテストは、タスク一覧の実装の条件にありますので、再確認してみましょう！
(例: 価格の範囲が¥300~¥9,999,999の間であること)

chirikuzu commented on spec/models/item_spec.rb:
```diff
@@ -0,0 +1,53 @@
+        @item.s_date_id = ""
+        @item.valid?
+        expect(@item.errors.full_messages).to include("SDate_id can't be blank")
+      end
```
active_hashを使用しているカラムについてはid:1が選択されている場合に出品できないことを検証しましょう。

chirikuzu reviewed (commented):
レビュー依頼ありがとうございます！
コメントしましたので、ご確認をお願いいたします！

5at05h1 committed:
エラーハンドリング、バリデーション

5at05h1 committed:
テストコード修正

chirikuzu commented on spec/factories/items.rb:
```diff
@@ -0,0 +1,14 @@
+    association :user
+    name            {"Apple"}
+    comment         {"ロゴ"}
+    category_id     {"家電・スマホ・カメラ"}
+    status_id       {"新品、未使用"}
+    shipping_id     {"送料込み(出品者負担)"}
+    delivery_id     {"大阪府"}
+    s_date_id       {"1~2日で発送"}
```
こちらはidが保存されるカラムになりますので、idの数字(1以外)を代入してあげましょう。

chirikuzu commented on spec/models/item_spec.rb:
```diff
@@ -0,0 +1,73 @@
+        @item.valid?
+        expect(@item.errors.full_messages).to include("Price can't be blank")
+      end
+    end
```
商品価格が299円以下のとき、10000000円以上のとき、画像が空のとき、それぞれ出品できないことをテストしましょう

chirikuzu reviewed (commented):
レビュー依頼ありがとうございます！
修正点確認させていただきました。
追加でコメントいたしましたので、ご確認お願いします！

5at05h1 committed:
コントローラー、テストコード修正

i-date commented on spec/factories/items.rb:
```diff
@@ -3,12 +3,12 @@
-    shipping_id     {"送料込み(出品者負担)"}
-    delivery_id     {"大阪府"}
-    s_date_id       {"1~2日で発送"}
+    category_id     {"2"}
+    status_id       {"2"}
+    shipping_id     {"2"}
+    delivery_id     {"10"}
+    s_date_id       {"2"}
```
idは数値になりますので、`"`で囲んで文字列にする必要はありません。
そのため、`"`は取り除きましょう。

i-date commented on spec/factories/items.rb:
```diff
@@ -3,12 +3,12 @@
+    shipping_id     {"2"}
+    delivery_id     {"10"}
+    s_date_id       {"2"}
     price           {"1000"}
```
11行目のpriceも同様に数値になりますので、`"`で囲んで文字列にする必要はありません。
そのため、`"`は取り除きましょう。

i-date reviewed (commented):
修正ありがとうございます！
追加補足コメントいたしましたので、ご確認をお願いいたします。

5at05h1 committed:
テストコード修正

Akinori0123 reviewed (commented):
レビュー依頼ありがとうございます。
LGTMです。実装お疲れ様でした。
引き続き実装頑張っていきましょう！

5at05h1 committed:
テストコード修正

5at05h1 committed:
JS非同期処理

入力された販売価格によって、販売手数料や販売利益が変わる非同期処理を忘れていたので記述しているのですが、やり方がよく分かっていません。
参考になるヒント等あれば教えて頂きたいです。

i-date commented:
> 入力された販売価格によって、販売手数料や販売利益が変わる非同期処理を忘れていたので記述しているのですが、やり方がよく分かっていません。
参考になるヒント等あれば教えて頂きたいです。

コメントいただきありがとうございます。
私たちコードレビューを行うメンターは、みなさまの実装したコードへ「指摘」をさせていただく存在となります。
そのため誠に恐れ入りますが、いただいた実装に関するコメントについては、まずはコネクトメンターへご質問いただけますでしょうか？
大変申し訳ないのですが、ご理解いただけますと幸いでございます。

5at05h1 committed:
JS非同期処理

Katsuno0814 commented on app/javascript/price.js:
```diff
@@ -1,3 +1,14 @@
+  const profit =document.getElementById("profit")
+  price.addEventListener('input', function(){
+    const integer = price.value
+    console.log(Math.floor(calculation));
+    const calculation = integer * 0.1
+    console.log(Math.floor(rieki));
```
こちらconsole.logは削除しておきましょう！

また、実際の画面や機能ができているか確認させていただきたいので、
キャプチャ動画をご提出いただけますでしょうか？

Katsuno0814 reviewed (commented):
レビュー依頼ありがとうございます！
修正点確認させていただきました。
追加でコメントいたしましたので、ご確認お願いします！

5at05h1 commented:
https://gyazo.com/0ce1a4e7eb4dc4d20011bb925176dd66

5at05h1 committed:
JS修正

miyagit commented on app/controllers/items_controller.rb:
```diff
@@ -0,0 +1,27 @@
+  before_action :authenticate_user!, only: [:new]
+
+  def index
+    # @items = Item.all
```
こちらコメントアウトしている理由は何でしょうか？

miyagit reviewed (commented):
レビュー依頼ありがとうございます。

5at05h1 commented:
今回はコメントアウトし、次回のプルリクで記述しましょうとレビューが来た為です。

iPhoneから送信

> 2020/09/29 12:52、Miyajima Yuya <notifications@github.com>のメール:
> 
> ﻿
> @miyagit commented on this pull request.
> 
> レビュー依頼ありがとうございます。
> 
> In app/controllers/items_controller.rb:
> 
> > @@ -0,0 +1,27 @@
> +class ItemsController < ApplicationController
> +  before_action :authenticate_user!, only: [:new]
> +
> +  def index
> +    # @items = Item.all
> こちらコメントアウトしている理由は何でしょうか？
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.

5at05h1 committed:
JS修正

i-date reviewed (approved):
修正とコメント返信ありがとうございます！
お疲れさまでした、LGTMとさせていただきます。
以降の実装も引き続きこの調子で進めていきましょう。応援しております！
